### DISCUSSION TITLE
Unable to compile for iOS development

### DISCUSSION THREAD

**Discussion by kasparv** [airsdk/Adobe-Runtime-Support#3378] [General]
Hey all!
We have a bunch of AS3-Flex-AIR-iOS apps that we have been maintaining for years in the Apple App Store and now I'd like to look into adding an in-app purchase feature. I've already gotten the InAppBilling ANE from Distriqt etc but I cannot seem to get development version iOS .ipa to compile correctly, even for the current codebase version of the app that compiled for Ad Hoc and iOS Distribution and uploaded to App Store without issues. 
I have created the iOS development certificate and iOS development profile, changed the `aps-environment` from `production`to `development`, but when I try to install the app on a device I get this error from Xcode:

> Error installing  '/Users/me/Developer/builds/iOS/app/MyAppMobile.ipa', ERROR: Error Domain=com.apple.dt.CoreDeviceError Code=3002 "Failed to install the app on the device." UserInfo={NSLocalizedDescription=Failed to install the app on the device., NSURL=file:///Users/me/Developer/builds/iOS/app/MyAppMobile.ipa, NSUnderlyingError=0x600002d5c570 {Error Domain=IXUserPresentableErrorDomain Code=14 "Unable to Install “My App”" UserInfo={NSUnderlyingError=0x600002d5d650 {Error Domain=MIInstallerErrorDomain Code=13 "Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.gjZKvy/extracted/Payload/MyAppMobile.app : 0xe8008015 (A valid provisioning profile for this executable was not found.)" UserInfo={FunctionName=+[MICodeSigningVerifier _validateSignatureAndCopyInfoForURL:withOptions:error:], LegacyErrorString=ApplicationVerificationFailed, SourceFileLine=79, LibMISErrorNumber=-402620395, NSLocalizedDescription=Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.gjZKvy/extracted/Payload/MyAppMobile.app : 0xe8008015 (A valid provisioning profile for this executable was not found.)}}, NSLocalizedDescription=Unable to Install “My App”, NSLocalizedRecoverySuggestion=Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.gjZKvy/extracted/Payload/MyAppMobile.app : 0xe8008015 (A valid provisioning profile for this executable was not found.), NSLocalizedFailureReason=This app cannot be installed because its integrity could not be verified.}}}
Domain: com.apple.dt.DVTCoreDevice
Code: -1
User Info: {
    DVTErrorCreationDateKey = "2024-07-25 16:41:52 +0000";
}
System Information
macOS Version 13.6.6 (Build 22G630)
Xcode 15.1 (22502) (Build 15C65)
Timestamp: 2024-07-25T19:41:52+03:00

I've tried with 51.0.1.4 and 50.2.3.8 AIR SDK overlaid onto Flex SDK.
I have verified that the developer certificate serial matches the certificate serial in the development profile, the testing devices are listed in the profile too. 
What could I be missing here?

**Comment by marchbold** [comment]
This can be caused by a mismatch of entries in your entitlements section vs what's in the profile you are using to package the app. 

Production apps have things like:

```xml
<key>beta-reports-active</key>
<true/>
<key>get-task-allow</key>
<false/>
<key>aps-environment</key>
<string>production</string>
```

Whereas your development version should have:

```xml
<key>get-task-allow</key>
<true/>
<key>aps-environment</key>
<string>development</string>
```

**Reply by kasparv** [reply]
Thank you for the directions!

> ```
> <key>get-task-allow</key>
> <true/>
> ```

Turned out fix was as simple as switching `get-task-allow` from `false` to `true` for the development version. 
Easy to miss when IDEs don't color the <![CDATA[ section...
